index.ts

```diff
@@ -14,18 +14,18 @@
  * to cleanly shut down before the process exits.
  */
 
-// TODO(rictic): convert this library to promises.
-
 /**
  * The type of a cleankill interrupt handler.
  */
-export type Handler = (doneHandling: () => void) => void;
+export type Handler = () => Promise<void>;
 
-let interruptHandlers: Handler[] = [];
+const interruptHandlers: Handler[] = [];
 
 /**
- * Register a handler to occur on SIGINT. All handlers are passed a callback,
- * and the process will be terminated once all handlers complete.
+ * Register a handler to occur on SIGINT. The handler must return a promise if
```

Comment:
Ah, I see. It seems that the two differences in that approach are in how errors propagate, and in the ergonomics.

It's kinda nice that cleankill can propagate errors from handlers up to the caller in `close`, which it couldn't do if `onInterrupt` returned a promise. The ergonomics is a bit of a wash I think. For some use cases I can see it being more convenient, for others, less.